tools/broadcasts.ts

```diff
@@ -0,0 +1,343 @@
+      }
+
+      return {
+        content: [
+          { type: 'text', text: 'Broadcast created successfully.' },
+          { type: 'text', text: `ID: ${response.data.id}` },
+          {
+            type: 'text',
+            text: "Don't bother telling the user the ID unless they ask for it.",
+          },
+        ],
+      };
+    },
+  );
+
+  server.tool(
+    'send-broadcast',
+    'Send a broadcast email by ID. You may optionally schedule the send.',
+    {
+      id: z.string().nonempty().describe('Broadcast ID'),
+      scheduledAt: z
+        .string()
+        .optional()
+        .describe(
+          'When to send the broadcast. Value may be in ISO 8601 format (e.g., 2024-08-05T11:52:01.858Z) or in natural language (e.g., "tomorrow at 10am", "in 2 hours", "next day at 9am PST", "Friday at 3pm ET"). If not provided, the broadcast will be sent immediately.',
+        ),
+    },
+    async ({ id, scheduledAt }) => {
+      console.error(`Debug - Sending broadcast with id: ${id}`);
+
+      const response = await resend.broadcasts.send(id, { scheduledAt });
+
+      if (response.error) {
+        throw new Error(
+          `Failed to send broadcast: ${JSON.stringify(response.error)}`,
+        );
+      }
+
+      return {
+        content: [
+          { type: 'text', text: 'Broadcast sent successfully.' },
+          { type: 'text', text: `ID: ${response.data.id}` },
+          {
+            type: 'text',
+            text: "Don't bother telling the user the ID unless they ask for it.",
+          },
+        ],
+      };
+    },
+  );
+
+  server.tool(
+    'list-broadcasts',
+    'List all broadcasts. Use this to find broadcast IDs or names.',
+    {},
+    async () => {
+      console.error('Debug - Listing broadcasts');
+
+      const response = await resend.broadcasts.list();
+
+      if (response.error) {
+        throw new Error(
+          `Failed to list broadcasts: ${JSON.stringify(response.error)}`,
+        );
+      }
+
+      const broadcasts = response.data.data;
+      return {
+        content: [
+          {
+            type: 'text',
+            text: `Found ${broadcasts.length} broadcast${broadcasts.length === 1 ? '' : 's'}${broadcasts.length === 0 ? '.' : ':'}`,
+          },
+          ...broadcasts.map(
+            ({
+              name,
+              id,
+              audience_id,
+              status,
+              created_at,
+              scheduled_at,
+              sent_at,
+            }) => ({
+              type: 'text' as const,
+              text: [
+                `ID: ${id}`,
+                `Name: ${name}`,
+                audience_id !== null && `Audience ID: ${audience_id}`,
+                `Status: ${status}`,
+                `Created at: ${created_at}`,
+                scheduled_at !== null && `Scheduled at: ${scheduled_at}`,
+                sent_at !== null && `Sent at: ${sent_at}`,
+              ]
+                .filter(Boolean)
+                .join('\n'),
```

Comment:
I've got grand plans to replace this with a fancy tagged template function that could be called like

```ts
text: lines`
  ID: ${id}
  Name: ${name}
  ${audience_id !== null && `Audience ID: ${audience_id}`}
  Status: ${status}
  Created at: ${created_at}
  ${scheduled_at !== null && `Scheduled at: ${scheduled_at}`}
  ${sent_at !== null && `Sent at: ${sent_at}`}
`,
```

But we'll see if I get around to it. This isn't too bad, I think.